### PR TITLE
Fix melting Native Token with foundry metadata

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account_manager.rs [hunk 1]
```diff
@@ -37,6 +38,7 @@ pub async fn new_account_manager(cli: AccountManagerCli) -> Result<(AccountManag
             let account_manager = AccountManager::builder()
                 .with_secret_manager(secret_manager)
                 .with_storage_path(&storage_path)
+                .with_coin_type(SHIMMER_COIN_TYPE)
```
Comment (Thoralf-M) [inline src/account_manager.rs]: Here it shouldn't be required

File: src/account_manager.rs [hunk 2]
```diff
@@ -56,6 +58,7 @@ pub async fn new_account_manager(cli: AccountManagerCli) -> Result<(AccountManag
             AccountManager::builder()
                 .with_secret_manager(secret_manager)
                 .with_storage_path(&storage_path)
+                .with_coin_type(SHIMMER_COIN_TYPE)
```
Comment (Thoralf-M) [inline src/account_manager.rs]: Here it shouldn't be required

File: src/account_manager.rs (hunk 1 above)
Comment (thibault-martinez) [reply src/account_manager.rs]: Right

File: src/account_manager.rs (hunk 2 above)
Comment (thibault-martinez) [reply src/account_manager.rs]: Looks like it's required

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**